src/types/xdg_activation_v1.zig

```diff
@@ -0,0 +1,51 @@
+const wlr = @import("../wlroots.zig");
+
+const wayland = @import("wayland");
+const wl = wayland.server.wl;
+
+pub const XdgActivationV1 = extern struct {
+    // token timeout in milliseconds (0 to disable)
+    token_timeout_msec: u32,
+
+    tokens: wl.list.Head(XdgActivationTokenV1, "link"),
+
+    events: extern struct {
+        destroy: wl.Signal(*XdgActivationV1),
+        request_activate: wl.Signal(*XdgActivationV1.RequestActivate.event),
+    },
+
+    // Private state
+    global: *wl.Global,
+
+    server_destroy: wl.Listener(*wl.Server),
+
+    pub const RequestActivate = extern struct {
+        event: extern struct {
+            activation: *wlr.XdgActivationV1,
+            token: *wlr.XdgActivationTokenV1,
+            surface: *wlr.Surface,
+        },
+    };
+
+    extern fn wlr_xdg_activation_v1_create(server: *wl.Server) ?*XdgActivationV1;
+    pub const create = wlr_xdg_activation_v1_create;
+};
+
+pub const XdgActivationTokenV1 = extern struct {
+    activation: *wlr.XdgActivationV1,
+    surface: ?*wlr.Surface,
+    seat: ?*wlr.Seat,
+
+    serial: ?u32, // Invalid if seat is null
```

Comment:
A `?u32` does not have the same ABI as a `u32` and I'm pretty sure won't even compile in an `extern struct` as it doesn't have a well defined memory layout.
```suggestion
    /// Invalid if seat is null
    serial: u32,
```